Lib/typing.py

```diff
@@ -215,38 +229,45 @@ def _remove_dups_flatten(parameters):
             params.extend(p[1:])
         else:
             params.append(p)
-    # Weed out strict duplicates, preserving the first of each occurrence.
-    all_params = set(params)
-    if len(all_params) < len(params):
-        new_params = []
-        for t in params:
-            if t in all_params:
-                new_params.append(t)
-                all_params.remove(t)
-        params = new_params
-        assert not all_params, all_params
+
+    return tuple(_deduplicate(params))
+
+
+def _flatten_literal_params(parameters):
+    """An internal helper for Literal creation: flatten Literals among parameters"""
+    params = []
+    for p in parameters:
+        if isinstance(p, _LiteralGenericAlias):
+            params.extend(p.__args__)
+        else:
```

Comment:
The [`_remove_dups_flatten`](https://github.com/python/cpython/blob/3.9/Lib/typing.py#L212) of `Union` contains this additional check:

```py
elif isinstance(p, tuple) and len(p) > 0 and p[0] is Union:
    params.extend(p[1:])
```

I wonder if such a check is relevant for `Literal` as well?

Reply:
Good catch, but I can realize what `Union` declaration case is covered by this if statement.
`Literal` tests cover all possible declaration mentioned at PEP.

Reply:
I don't understand why there would be a nested tuple in Union parameters. Can you show an example that ever triggers that check?

Reply:
I have just run `test_typing` and no test didn't enter this if statement.
Looks like it's a dead code that we can remove.

Reply:
Well at least we shouldn’t replicate it for Literal.

Reply:
Makes sense, I will resolve this conversation.